catalog/core/catalog-core-api/src/main/java/ddf/catalog/plugin/PrePolicyPlugin.java

```diff
@@ -0,0 +1,104 @@
+ */
+package ddf.catalog.plugin;
+
+import java.util.Map;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.UpdateRequest;
+
+/**
+ * The {@link PrePolicyPlugin} extension point allows for request/response processing before any
+ * policies are implemented at the very beginning of the plugin chain.
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ */
+public interface PrePolicyPlugin {
+
+    /**
+     * @param input the {@link CreateRequest} to process
+     * @return the value of the processed {@link CreateRequest} to pass to the next {@link AccessPlugin}
+     * @throws StopProcessingException thrown to halt processing when a critical issue occurs during processing. This is
+     *                                 intended to prevent other plugins from processing as well.
+     */
+    CreateRequest processPreCreate(CreateRequest input) throws StopProcessingException;
+
+    /**
+     * @param input             the {@link UpdateRequest} to process
+     * @param existingMetacards the Map of {@link Metacard}s that currently exist
+     * @return the value of the processed {@link UpdateRequest} to pass to the next {@link AccessPlugin}
+     * @throws StopProcessingException thrown to halt processing when a critical issue occurs during processing. This is
+     *                                 intended to prevent other plugins from processing as well.
+     */
+    UpdateRequest processPreUpdate(UpdateRequest input, Map<String, Metacard> existingMetacards)
+            throws StopProcessingException;
+
+    /**
+     * @param input the {@link DeleteRequest} to process
+     * @return the value of the processed {@link DeleteRequest} to pass to the next {@link AccessPlugin}
+     * @throws StopProcessingException thrown to halt processing when a critical issue occurs during processing. This is
+     *                                 intended to prevent other plugins from processing as well.
+     */
+    DeleteRequest processPreDelete(DeleteRequest input) throws StopProcessingException;
+
+    /**
+     * @param input the {@link DeleteResponse} to process
+     * @return the value of the processed {@link DeleteResponse} to pass to the next {@link AccessPlugin}
+     * @throws StopProcessingException thrown to halt processing when a critical issue occurs during processing. This is
+     *                                 intended to prevent other plugins from processing as well.
+     */
+    DeleteResponse processPostDelete(DeleteResponse input) throws StopProcessingException;
```

Comment:
You provide methods for both pre- and post-delete but only for pre-create and pre-update. Is that intentional?